### ISSUE TITLE
[feature] Implement request.getCookie(name)

### ISSUE THREAD

**Issue by uasan** [uNetworking/uWebSockets.js#399]
Hey.
You have implemented the [request.getQuery(name)](
https://github.com/uNetworking/uWebSockets/issues/1091#issuecomment-700032011) method with optional argument.

Often need a method `request.getCookie(name)`.
The best i could come up with in JS:
```JS
import { parse } from 'querystring';

parse(request.getHeader('cookie'), '; ')

```
But, parsing cookies in JS is also not productive.
It will be great if the `getCookie(name)` method appears in request object and is as efficient as `getQuery(name)`.
Thanks.

**Comment by hst-m** [comment]
I posted this in other issue, what do you think:
```javascript
const cookieRegexObj={}
const cookieRegex=name=>cookieRegexObj[name]||(cookieRegexObj[name]=new RegExp(`(^|;)\\s*${name}\\s*=\\s*([^;]+)`))
const getCookie=(req,name)=>(req.cookies||(req.cookies=req.getHeader('cookie'))).match(cookieRegex(name))?.[2]

// example
const cookie=getCookie(req,'cookieName')
```
it saves result of `req.getHeader('cookie')` and cookie regex for future lookups, I guess only question is if is as fast as anything native, but for JS I recommend this instead of external library because this is all you need

**Comment by hst-m** [comment]
this is what I use:
```javascript
const cookieRegexObj={}
const cookieRegex=name=>cookieRegexObj[name]||(cookieRegexObj[name]=new RegExp(`(^|;)\\s*${name}\\s*=\\s*([^;]+)`))
const getCookie=(req,name)=>(req.cookies||(req.cookies=req.getHeader('cookie'))).match(cookieRegex(name))?.[2]

// example
const cookie=getCookie(req,'cookieName')
```
I just tested this vs `querystring.parse` method, I show 20K nanoseconds for my `getCookie` and 60K nanosec for `querystring.parse`, tested with getting multiple largish cookies in same request and also single cookie request

**Comment by uasan** [comment]
`querystring.parse` - this method parses all cookies from the header line and `decodeURIComponent()` their values, I think this is the time difference

**Comment by alexhultman-2fa** [comment]
This could make sense if getCookie is often used. It looks to be similar to getQuery - but there is no docs saying it should be URI decoded (some say it should be base64).

**Comment by uasan** [comment]
> no docs saying it should be URI decoded (some say it should be base64).

Yes, this is not part of the specification, this is the actual implementation for encoding special characters, but for many business logic an is normal `raw` implementation - without decoding
